### ISSUE TITLE
Unable to use camera or microphone

### ISSUE THREAD

**Issue by Ep0chalypse** [bromite/bromite#1549]
### Bromite version

Version: `95.0.4638.78`
Arch: `arm`
Android version: `12`
Device model: `Pixel 5a`

### Flags

These are the flags changed under `chrome://flags`:
None changed

### Is this bug about the SystemWebView?

No

### Is the bug reproducible with latest version?

Yes

### Can the bug be reproduced with corresponding Chromium version?

No

### Is this bug happening in an incognito tab?
Yes

### Allow ads for the site; is the bug caused by the adblocker?
No

### Is the bug a crash?

No

```
<paste here logcat dump individuated with 'adb logcat | grep chromium'>
```

### Describe the bug

When I navigate to a website(such as https://webrtc.github.io/samples/src/content/devices/input-output/ or https://doxy.me/precall-test ) that uses the phones camera or microphone, neither work. I have disabled and re-enabled all Bromite permissions in androids permission manager. I have tried to clear the cache and data for the app. I have reinstalled the app. None of these steps have solved this issue. Curiously enough, in Android 12, they added a new notification that is a green dot in the upper right of the notification bar that shows up when a app is accessing the camera or microphone. When I go to those website and allow camera and microphone, the green dot appears, but neither the front or back camera show video and no audio is recorded and played back.

### Steps to reproduce the bug

1. Go to https://webrtc.github.io/samples/src/content/devices/input-output/ or https://doxy.me/precall-test
2. Allow access to camera and microphone

Do not write "any website", specify which URLs can be used to reproduce the issue.

### Expected behavior

Expect to see video from devices camera and audio from the microphone.

**Comment by kokukyocho** [comment]
Hello @Ep0chalypse!

I am a bot :robot:

Thanks for submitting this issue!
I noticed that it is missing the template, please edit the issue to match the template for either a [bug](https://github.com/bromite/bromite/blob/master/.github/ISSUE_TEMPLATE/bug_report.md) or a [feature request](https://github.com/bromite/bromite/blob/master/.github/ISSUE_TEMPLATE/feature_request.md); the issue will be automatically re-opened afterwards.

If this is not an issue nor a feature request and you want to start a discussion then you can use [GitHub Discussions](https://github.com/bromite/bromite/discussions).

**Tip**: if you use GitHub in Desktop mode the template will be proposed to you when submitting issues.

**Comment by csagan5** [comment]
This is odd, I wonder when it started happening; @uazo @wchen342 can you think of any reason why this could be happening?

**Comment by uazo** [comment]
> can you think of any reason why this could be happening?

Yes, it's the Unified Autoplay policy.
if you put it to `FEATURE_ENABLED_BY_DEFAULT` it works correctly.

**Comment by wchen342** [comment]
Interesting, I have never seen this. @uazo You mean [this](https://blog.chromium.org/2017/09/unified-autoplay.html)? Seems like an old feature and it is more like WebRTC specific but unrelated to Android 12?

**Comment by uazo** [comment]
we discussed it here https://github.com/bromite/bromite/issues/791#issuecomment-724140535

**Comment by csagan5** [comment]
> > can you think of any reason why this could be happening?
> 
> Yes, it's the Unified Autoplay policy. if you put it to `FEATURE_ENABLED_BY_DEFAULT` it works correctly.

Then if the website requires some user interaction webcam and microphone will work?

**Comment by uazo** [comment]
> Then if the website requires some user interaction webcam and microphone will work?
> Perhaps they removed the requirement for the interaction because there is already the prompt about allowing camera/microphone.

yes, tried it, I can confirm it, you must give permission to use webcam and microphone.
and even the autoplay block continues to work.

**Comment by csagan5** [comment]
Then these websites do not work because they expect automatically working microphone and video without user interaction.

**Comment by uazo** [comment]
it looks like this, and it was a concern we had already discussed by deciding to wait for the first bug.

**Comment by csagan5** [comment]
This is not the first bug reported about this; the only thing we could do is stretch the definition of user interaction to include "giving permission", since the playback at least waits for the permission to be granted (is this true?).

Original issue: #804

Playback cannot happen without the use explicit approval though.

**Comment by uazo** [comment]
> the only thing we could do is stretch the definition of user interaction to include "giving permission", since the playback at least waits for the permission to be granted (is this true?).

yes, that's true, but only the first time and only if the browser does not have access permissions granted by the s.o.
actually, the browser checks the content settings (audio/microphone) but without any request through ui and, once granted, the site has the permission to use even without displaying anything.

I make two proposals:
1) create a new setting "request permissions at each session" to request access each time the application is restarted (or after a specific timeout, say, 4 hours).
2) change the content settings to work "per session"

between the two, the first is simpler because theoretically it would be enough for me to intercept the current authorization call and submit it to a check; to allow operation per session it would be enough for me to use a variable.
the second is obviously more complex...

**Comment by csagan5** [comment]
I think it is correct to diverge from the default that Chromium and forks do; what is the correct way to see this working in Bromite? We should create a working example on the Bromite website.

**Comment by jeffersoncheang** [comment]
Hi. I'm having the exact same issue using the latest Bromite version (previous versions have the same issue).
Arch: arm
Android version: 11
Device model: OnePlus 8T

**Comment by thomas725** [comment]
Stumbled upon the same issue. Also on Android 11 latest bromite version installed through "FFUpdater" app found on F-Droid.

**Comment by csagan5** [comment]
@uazo also here, you went ahead and created the patch, but the discussion was not concluded. Since the patch has not its separate PR, there has been no discussion on the implementation either (everything is in #1752); there might be security issues there.

**Comment by uazo** [comment]
> but the discussion was not concluded.

sure, let's talk about it. among other things maybe I found a better way when I saw the code of the client hints: basically the content settings can work per "session". I have to try it.

> Since the patch has not its separate PR

we can fix that, we just have to wait for the versions to be aligned

> there might be security issues there.

let me understand what I should check.

**Comment by csagan5** [comment]
> > but the discussion was not concluded.
> 
> sure, let's talk about it. among other things maybe I found a better way when I saw the code of the client hints: basically the content settings can work per "session". I have to try it.

Sound cool!

> > there might be security issues there.
> 
> let me understand what I should check.

For example: imagine a scenario where everything is granted (because of a bug in the new patch) even when it should not. But let's discuss it in the PR for the new patch, no hurry.

**Comment by csagan5** [comment]
@uazo you can now open a PR for this since patches for v99 are ready.

**Comment by csagan5** [comment]
Possibly related: `kQuietNotificationPrompts`

**Comment by uazo** [comment]
also `kOneTimeGeolocationPermission`

**Comment by csagan5** [comment]
@uazo still planning to open a PR for this?

**Comment by uazo** [comment]
> still planning to open a PR for this?

Yes, with some time

**Comment by csagan5** [comment]
Sure, no hurry; I am asking as I regularly go through all open issues.

**Comment by csagan5** [comment]
Fixed in `100.0.4896.83`.
